test/integ-new/main_new_test.go

```diff
@@ -0,0 +1,212 @@
+	Req string
+	Sub string
+}
+
+type testValidationResult struct {
+	Breakglass     bool
+	Justifications []v0.Justification
+	Claims         testClaim
+}
+
+func TestMain(m *testing.M) {
+	os.Exit(func() int {
+		ctx := context.Background()
+
+		if strings.ToLower(os.Getenv("TEST_INTEGRATION")) != "true" {
+			log.Printf("skipping (not integration)")
+			// Not integration test. Exit.
+			return 0
+		}
+
+		// set up global test config.
+		c, err := newTestConfig(ctx)
+		if err != nil {
+			log.Printf("Failed to parse integration test config: %v", err)
+			return 2
+		}
+		cfg = c
+
+		return m.Run()
+	}())
+}
+
+func TestAPIAndPublicKeyService(t *testing.T) {
+	t.Parallel()
+	ts := time.Now().UTC()
+	cases := []struct {
+		name                     string
+		isBreakglass             bool
+		justification            string
+		wantTestValidationResult *testValidationResult
+	}{
+		{
+			name:          "none_breakglass",
+			justification: "issues/12345",
+			isBreakglass:  false,
+			wantTestValidationResult: &testValidationResult{
+				Breakglass: false,
+				Justifications: []v0.Justification{
+					{
+						Category: "explanation",
+						Value:    "issues/12345",
+					},
+				},
+				Claims: testClaim{
+					Aud: []string{"dev.abcxyz.jvs"},
+					Exp: ts.Add(testTTLTime).UTC().Format(timestampFormat),
+					Iat: ts.Format(timestampFormat),
+					Iss: "jvs.abcxyz.dev",
+					Req: cfg.ServiceAccount,
+					Sub: cfg.ServiceAccount,
+				},
+			},
+		},
+		{
+			name:          "breakglass",
+			justification: "issues/12345",
+			isBreakglass:  true,
+			wantTestValidationResult: &testValidationResult{
+				Breakglass: true,
+				Justifications: []v0.Justification{
+					{
+						Category: "breakglass",
+						Value:    "issues/12345",
+					},
+				},
+				Claims: testClaim{
+					Aud: []string{"dev.abcxyz.jvs"},
+					Exp: ts.Add(testTTLTime).UTC().Format(timestampFormat),
+					Iat: ts.Format(timestampFormat),
+					Iss: "jvsctl",
+				},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			var createCmd cli.TokenCreateCommand
+			_, stdout, _ := createCmd.Pipe()
+
+			createTokenArgs := []string{"-server", cfg.APIServer, "-justification", tc.justification, "-ttl", testTTLString, "--auth-token", cfg.IDToken}
+			if tc.isBreakglass {
+				createTokenArgs = append(createTokenArgs, "-breakglass")
+			}
+
+			if err := createCmd.Run(ctx, createTokenArgs); err != nil {
+				t.Fatal(err)
+			}
+
+			token := stdout.String()
+
+			validateTokenArgs := []string{"-token", token, "-jwks-endpoint", cfg.JWTEndpoint, "--format", "json"}
+
+			var validateCmd cli.TokenValidateCommand
+			_, stdout, _ = validateCmd.Pipe()
+
+			if err := validateCmd.Run(ctx, validateTokenArgs); err != nil {
+				t.Fatal(err)
+			}
+
+			gotTestValidationResult := &testValidationResult{}
+			if err := json.Unmarshal(stdout.Bytes(), gotTestValidationResult); err != nil {
```

Comment:
What happens if the json string has some fields that are not found in the gotTestValidationResult struct? Maybe unmarshal it to a map is better.

Reply:
Or should we use jwt.Token instead?

Reply:
Did a quick check, it turns out we are using https://github.com/abcxyz/jvs/blob/9a2eff640dced585729be84a7a53f71869bd61d4/pkg/formatter/formatter.go#L36 for the json formatter.

Reply:
I thought more about this. For context, this formatted structure _is_ defined here: https://github.com/abcxyz/jvs/blob/main/pkg/formatter/formatter.go#L36 except that it's not public.
However, I don't think we should run the test based on the output. The output format is simply a representation of the input data and it can be fully covered in unit test.
What we really care here is whether the running service can give us token and validate token. I recommend updating the test:

1. Run first command to create a token. Ensure success.
2. Run second command to validate the token. Ensure success. Don't check the output.
3. Parse the token to ensure required claims are there.

The only difference is the 3rd step. Instead of checking the output which is up to change, we check the token directly.

Reply:
Ok, so we are probably fine here.

Reply:
To clarify, it is doing the three things in a single test func, or do you mean to split the test func into 2, one for create token and one for validate token?